django/db/models/aggregates.py

```diff
@@ -211,3 +213,15 @@ def __init__(self, expression, sample=False, **extra):
 
     def _get_repr_options(self):
         return {**super()._get_repr_options(), "sample": self.function == "VAR_SAMP"}
+
+
+class JSONArrayAgg(Aggregate):
+    name = "JSONArrayAgg"
+    output_field = JSONField()
+    arity = 1
+
+    def __init__(self, expression, **extra):
+        self.function = (
+            "JSON_GROUP_ARRAY" if connection.vendor == "sqlite" else "JSON_ARRAYAGG"
+        )
```

Comment:
👋 @lufafajoshua!

You can't use `connection` for that since some projects might have `DATABASES` entries for different backends.

Refer to [the `as_sql` documentation](https://docs.djangoproject.com/en/5.0/ref/models/expressions/#django.db.models.Func.as_sql) particularly the part about `as_vendor` and refer to [examples such as `Greatest`](https://github.com/django/django/blob/dfd63ff43408e7901cc214b0482a7f844244d439/django/db/models/functions/comparison.py#L134-L143) to get an idea of how to achieve what you're after.